gapic/src/main/com/google/gapid/views/DeviceDialog.java

```diff
@@ -0,0 +1,303 @@
+      deviceCombo.setSelection(null);
+      deviceCombo.getCombo().notifyListeners(SWT.Selection, new Event());
+      deviceLoader.stopLoading();
+
+    }
+
+    private Device.Instance getSelectedDevice() {
+      IStructuredSelection sel = deviceCombo.getStructuredSelection();
+      return sel.isEmpty() ? null : (Device.Instance) sel.getFirstElement();
+    }
+
+    private void runValidationCheck(Device.Instance deviceInstance) {
+      if (deviceInstance == null) {
+        validationStatusLoader.setVisible(false);
+        validationStatusText.setVisible(false);
+        return;
+      }
+      validationStatusLoader.setVisible(true);
+      validationStatusText.setVisible(true);
+      // We need a DeviceCaptureInfo to do validation.
+      DeviceCaptureInfo dev =
+          new DeviceCaptureInfo(Paths.device(deviceInstance.getID()), deviceInstance, null, null);
+      setValidationStatus(models.devices.getValidationStatus(dev));
+      if (!models.devices.getValidationStatus(dev).passed) {
+        validationStatusLoader.startLoading();
+        validationStatusText.setText("Device is being validated");
+        rpcController.start().listen(models.devices.validateDevice(dev),
+            new UiErrorCallback<DeviceValidationResult, DeviceValidationResult, DeviceValidationResult>(
+                validationStatusLoader, LOG) {
+              @Override
+              protected ResultOrError<DeviceValidationResult, DeviceValidationResult> onRpcThread(
+                  Rpc.Result<DeviceValidationResult> response)
+                  throws RpcException, ExecutionException {
+                try {
+                  return success(response.get());
+                } catch (RpcException | ExecutionException e) {
+                  throttleLogRpcError(LOG, "LoadData error", e);
+                  return error(null);
+                }
+              }
+
+              @Override
+              protected void onUiThreadSuccess(DeviceValidationResult result) {
+                setValidationStatus(result);
+              }
+
+              @Override
+              protected void onUiThreadError(DeviceValidationResult result) {
+                LOG.log(WARNING, "UI thread error while validating device");
+                setValidationStatus(result);
+              }
+            });
+      }
+    }
+
+    protected void setValidationStatus(DeviceValidationResult result) {
+      if (result.skipped) {
+        validationStatusLoader.updateStatus(true);
+        validationStatusLoader.stopLoading();
+        validationStatusText.setText("Validation skipped.");
+        validationPassed = true;
+      } else {
+        validationStatusLoader.updateStatus(result.passed);
+        validationStatusText.setText("Validation "
+            + (result.passed ? "Passed." : "Failed. " + Messages.VALIDATION_FAILED_LANDING_PAGE));
+        validationStatusLoader.stopLoading();
+        validationPassed = result.passed;
```

Comment:
OK button enabled state needs to update when validation succeeds

Reply:
done